server/connector.go

```diff
@@ -189,6 +191,26 @@ func (c *connectorImpl) findAndConnectBackend(ctx context.Context, frontendConn
 	c.metrics.ActiveConnections.Add(1)
 	defer c.metrics.ActiveConnections.Add(-1)
 
+	// PROXY protocol implementation
+
+	remoteAddr, err := net.ResolveIPAddr("ip", strings.Split(backendHostPort, ":")[0])
+	if err != nil {
+		// handle error
+	}
+
+	header := &proxyproto.Header{
+		Version:           2,
+		Command:           proxyproto.PROXY,
+		TransportProtocol: proxyproto.TCPv4,
+		SourceAddr:        clientAddr,
+		DestinationAddr:   remoteAddr,
+	}
+
+	_, err = header.WriteTo(backendConn)
+	if err != nil {
+		// handle error
```

Comment:
Borrow from one of the blocks below and do something like:

```suggestion
		logrus.
			WithError(err).
			WithField("client", clientAddr).
			Error("Failed to write proxy header")
		c.metrics.Errors.With("type", "write").Add(1)
		_ := backendConn.Close()
		return
```